panther_lights/src/driver_node.cpp

```diff
@@ -0,0 +1,176 @@
+// Copyright 2023 Husarion sp. z o.o.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <panther_lights/driver_node.hpp>
+
+#include <cstdint>
+#include <filesystem>
+#include <limits>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <gpiod.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <image_transport/image_transport.hpp>
+#include <sensor_msgs/image_encodings.hpp>
+
+#include <panther_msgs/srv/set_led_brightness.hpp>
+
+#include <panther_lights/apa102.hpp>
+
+namespace panther_lights
+{
+
+using std::placeholders::_1;
+using std::placeholders::_2;
+
+DriverNode::DriverNode(const std::string & node_name, const rclcpp::NodeOptions & options)
+: Node(node_name, options), front_panel_("/dev/spidev0.0"), rear_panel_("/dev/spidev0.1")
+{
+  rclcpp::on_shutdown(std::bind(&DriverNode::OnShutdown, this));
+
+  this->declare_parameter<double>("global_brightness", 1.0);
+  this->declare_parameter<double>("frame_timeout", 0.1);
+  this->declare_parameter<int>("num_led", 46);
+
+  RCLCPP_INFO(this->get_logger(), "Node started");
+}
+
+void DriverNode::Initialize()
+{
+  it_ = std::make_shared<image_transport::ImageTransport>(this->shared_from_this());
+
+  const double global_brightness = this->get_parameter("global_brightness").as_double();
```

Comment:
```suggestion
  const flaot global_brightness = static_cast<float>(this->get_parameter("global_brightness").as_double());
```
The service uses float so passing a double does not make much sense